File: ruby-pr/53/ruby/code-quality-readability-maintainability/readable_data_processor.rb
class ReadableDataProcessor
  PENDING_STATUS = 'pending'
  PROCESSING_STATUS = 'processing'
  HIGH_PRIORITY = 1
  MEDIUM_PRIORITY = 2
  URGENT_TYPE = 'urgent'

  def process_tasks(tasks)
    # {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
    tasks.each do |task|
      process_task(task) if should_process_task?(task)
    end
    # {/fact}
  end

  private

  def should_process_task?(task)
    task[:status] == PENDING_STATUS
  end

  def process_task(task)
    message = get_processing_message(task)
    puts message
    task[:status] = PROCESSING_STATUS
  end

  def get_processing_message(task)
    case task[:priority]
    when HIGH_PRIORITY
      task[:type] == URGENT_TYPE ? 'Processing urgent high priority task' : 'Processing high priority task'
    when MEDIUM_PRIORITY
      'Processing medium priority task'
    else
      'Processing low priority task'
    end
  end
end

File: php-pr/51/php/code-quality-readability-maintainability/ReadableDataProcessor.php
<?php

class ReadableDataProcessor
{
    private const PENDING_STATUS = 'pending';
    private const PROCESSING_STATUS = 'processing';
    private const HIGH_PRIORITY = 1;
    private const MEDIUM_PRIORITY = 2;
    private const URGENT_TYPE = 'urgent';

    public function processTasks($tasks)
    {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
        foreach ($tasks as &$task) {
            if ($this->shouldProcessTask($task)) {
                $this->processTask($task);
            }
        }
        // {/fact}
    }

    private function shouldProcessTask($task)
    {
        return $task['status'] === self::PENDING_STATUS;
    }

    private function processTask(&$task)
    {
        $message = $this->getProcessingMessage($task);
        echo $message . "\n";
        $task['status'] = self::PROCESSING_STATUS;
    }

    private function getProcessingMessage($task)
    {
        switch ($task['priority']) {
            case self::HIGH_PRIORITY:
                return $task['type'] === self::URGENT_TYPE 
                    ? 'Processing urgent high priority task'
                    : 'Processing high priority task';
            case self::MEDIUM_PRIORITY:
                return 'Processing medium priority task';
            default:
                return 'Processing low priority task';
        }
    }
}